src/api/lib/rabbitmq_bus.rb

```diff
@@ -1,29 +1,42 @@
 class RabbitmqBus
+  cattr_accessor :connection, :exchange
+
   def self.send_to_bus(channel, data)
     publish("#{Configuration.amqp_namespace}.#{channel}", data)
   rescue => e
     Rails.logger.error "Publishing to RabbitMQ failed: #{e.message}"
     Airbrake.notify(e)
+    # reconnect on next message
+    RabbitmqBus.connection.stop
+    RabbitmqBus.exchange = nil
   end
 
   def self.publish(event_routing_key, event_payload)
     return unless CONFIG['amqp_options']
-    start_connection
+    start_connection unless connection && connection.open?
+    wrapped_exchange.publish(event_payload, routing_key: event_routing_key)
+  end
 
-    $rabbitmq_exchange.publish(event_payload, routing_key: event_routing_key)
+  def self.wrapped_exchange
+    connection.start
+    rabbitmq_channel = connection.create_channel
+    RabbitmqBus.exchange = if CONFIG['amqp_exchange_name']
+      rabbitmq_channel.exchange(CONFIG['amqp_exchange_name'], CONFIG['amqp_exchange_options'].try(:symbolize_keys) || {})
+    else
+      # default exchanges don't publish
+      # https://github.com/arempe93/bunny-mock/pull/25
```

Comment:
:thought_balloon: so you're going to update this comment whenever the issue changes? ;-)

Reply:
No, whenever the pull request is merged, we can increase the coverage to this.